### PR TITLE
Adding new property zone_redundant to azurerm_container_app_environment

### PR DESCRIPTION
**Pull request by matt-lethargic**
fixes [#2053](https://github.com/hashicorp/terraform-provider-azurerm/issues/20538)

adds zone_redundant property to azurerm_container_app_environment

### CODE REVIEW

**Comment by matt-lethargic** [comment]
Hey @jackofallops I see you've picked this up. I just looked at the files changed and realised that it thinks I've changed every line. I'm not sure how this happened as I followed the contributing instructions. Is there anything I can do to make this easier to review?

File: internal/services/containerapps/container_app_environment_resource_test.go [hunk 1]
```diff
@@ -1,224 +1,226 @@
...
+
+  internal_load_balancer_enabled = true
+
+  tags = {
+    Foo    = "Bar"
+    secret = "sauce"
+  }
+
+  zone_redundant = true
+}
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource_test.go:160]: Can we move this up to with the other properties?

```suggestion
resource "azurerm_container_app_environment" "test" {
  name                       = "acctest-CAEnv%[2]d"
  resource_group_name        = azurerm_resource_group.test.name
  location                   = azurerm_resource_group.test.location
  log_analytics_workspace_id = azurerm_log_analytics_workspace.test.id
  zone_redundant             = true
  
  infrastructure_subnet_id = azurerm_subnet.control.id

  internal_load_balancer_enabled = true

  tags = {
    Foo    = "Bar"
    secret = "sauce"
  }
}

```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @matt-lethargic - Thanks for this PR. There's a few comments below that need to be addressed, then we can run the GHA's and I can put it through acceptance testing.

Thanks!

**Comment by matt-lethargic** [comment]
@jackofallops Thanks for the feedback, updated

**Comment by matt-lethargic** [comment]
@jackofallops a friendly bump ;-)

**Comment by jackofallops** [comment]
> @jackofallops a friendly bump ;-)

Hi @matt-lethargic - apologies, I've been out on PTO for a while. Would you be able to rebase against `main` to resolve the conflicts, and I'll finish up review and run the tests?

Thanks!

**Comment by jackofallops** [comment]
Hi @matt-lethargic - Sincerest apologies, in attempting to rebase your branch to resolve the conflicts, I've typo'd the fetch and overwritten your commits :( I'm going to have to close this PR and I'll open a new one and I'll mark the commits as co-authored by you.  Again, I can only apologise!

**Pull request closed without merging**